http/backup_service.go

```diff
@@ -106,6 +131,65 @@ func (h *BackupHandler) handleBackupShard(w http.ResponseWriter, r *http.Request
 	}
 }
 
+func (h *BackupHandler) handleBackupMetadata(w http.ResponseWriter, r *http.Request) {
+	span, r := tracing.ExtractFromHTTPRequest(r, "BackupHandler.handleBackupMetadata")
+	defer span.Finish()
+
+	ctx := r.Context()
+
+	baseName := time.Now().UTC().Format(influxdb.BackupFilenamePattern)
+
+	formWriter := multipart.NewWriter(w)
+	w.Header().Set("Content-Type", formWriter.FormDataContentType())
+
+	parts := []struct {
+		fieldname string
+		filename  string
+		writeFn   func(io.Writer) error
+	}{
+		{
+			"bolt",
+			fmt.Sprintf("%s.bolt", baseName),
+			func(fw io.Writer) error {
+				return h.BackupService.BackupKVStore(ctx, fw)
+			},
+		},
+		{
+			"sqlite",
+			fmt.Sprintf("%s.sqlite", baseName),
+			func(fw io.Writer) error {
+				return h.SqlBackupRestoreService.BackupSqlStore(ctx, fw)
```

Comment:
Is there any way we could take a lock across both KV and SQL before running these commands? If not there's still a window where inconsistencies could creep in.

Reply:
I added locking mechanisms. It required a fair amount of modifications, but seems to work.